uw-frame-components/portal/main/controllers.js

```diff
@@ -72,19 +72,25 @@ define(['angular','require'], function(angular, require) {
 
   /* Username */
   app.controller('SessionCheckController', [ '$scope', 'mainService', 'NAMES', 'FOOTER_URLS', '$rootScope', function($scope, mainService, NAMES, FOOTER_URLS, $rootScope) {
-    var that = this;
-    that.user = [];
+    var vm = this;
+    vm.user = [];
+    vm.firstLetter = '';
 
     $scope.FOOTER_URLS = FOOTER_URLS;
     $scope.usernameOptionOpen = false;
 
     mainService.getUser().then(function(result){
-      that.user = result;
-      //check if is guest
-      if (NAMES.guestUserName && that.user && that.user.userName === NAMES.guestUserName)
+      vm.user = result;
+      // Check if is guest
+      if (NAMES.guestUserName && vm.user && vm.user.userName === NAMES.guestUserName) {
         $rootScope.GuestMode = true;
-
+      } else {
+        // Get first letter of first name or display name
+        var username = vm.user.firstName ? vm.user.firstName : vm.user.displayName;
+        vm.firstLetter = username.substring(0,1);
```

Comment:
Could we make this fail safe to, say, "?" in the case where `username` ended up `nil` or `""`.

It'll probably never actually happen, but user attributes, they can get weird.